src/olympia/addons/views.py

```diff
@@ -209,13 +209,11 @@ def check_object_permissions(self, request, obj):
         `is_disabled_by_mozilla` to the exception being thrown so that clients
         can tell the difference between a 401/403 returned because an add-on
         has been disabled by their developer or something else.
-
-        On top of this, can also raise a 451 if the add-on is not available
-        because of regional restrictions - no additional detail is available
-        in that case.
         """
         try:
             super(AddonViewSet, self).check_object_permissions(request, obj)
+        except UnavailableForLegalReasons as exec_451:
+            raise exec_451
```

Comment:
This is because `UnavailableForLegalReasons` subclasses `exceptions.APIException` but we don't need the extra `is_disabled` data when it's a 451.